spotbugs/src/main/java/edu/umd/cs/findbugs/sarif/Location.java

```diff
@@ -137,6 +137,19 @@ JSONObject toJSONObject() {
                     throw new UncheckedIOException(e);
                 }
             });
+
+            if (location.isPresent()) {
+                return location;
+            }
+
+            try {
+                String path = bugAnnotation.format("full", classAnnotation);
```

Comment:
added a fallback in case we don't have a location yet. This is printing the same thing that were being used in logicalLocation.fullyqualifiedname